### PR TITLE
Remove legacy json lib

### LINKED ISSUE #379: elementbrowser is broken
**Which resource(s) have this problem?**

elementbrowser

**To Reproduce**

1 - start the `elementbrowser` resource
2 - open http://server_ip:http_port
3 - select the "Element browser" menu option

**Expected behavior**

it should work 😄

**Screenshots**

image
image

**Version**

MTA:SA Server 1.5.9-release-21048

**Additional context**

I haven't checked the code, but I belive its the same thing @ArranTuna fixed in [this commit](https://github.com/multitheftauto/mtasa-resources/commit/37348cecb60a6c999c316398a17f866d54569953)

### PR DESCRIPTION
**Pull request by 4O4**
Last modifications made to `json.js` lib during webmap upgrade ([[web]/ajax/json.js](https://github.com/multitheftauto/mtasa-resources/commit/8e2b12b6d012c89a063a8c5e9873454f8d3ab7e3#diff-adfe7968957f2a79a6ec34240a7f5d87186a3b1a64b8c52baa91f1f78c5c7a29)) broke other resources. This lib however is no longer needed as there have been native JSON functions in Javascript for ages already.

This PR reverts a workaround previously introduced in #358, removes the old lib and replaces it with native JSON functions. 

This also involved a slight refactoring. A bit of more modern code was introduced (`class`es, `startsWith`). I'm assuming we don't care about ancient browsers anymore and ES2015 is a safe target for any new JS code.

Closes #379 #358

### CODE REVIEW

File: [web]/ajax/ajax.js [hunk 1]
```diff
@@ -115,18 +115,16 @@ function getKeyCode ( e )
...
-		for ( i = 0; i < this.elements.length; i++ )
-		{
+class ElementManager {
+	constructor() {
+		this.elements = new Array();
+	}
+	
+	get(id) {
+		for (let i = 0; i < this.elements.length; i++) {
 			var element = this.elements[i];
```
Comment (patrikjuvonen) [inline [web]/ajax/ajax.js:125]: Prefer [for...of](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of).

```suggestion
		for (const element of this.elements) {

```

File: [web]/ajax/ajax.js [hunk 2]
```diff
@@ -137,18 +135,16 @@ function ElementManager() {
...
+	}
 
-function ResourceManager() {
-	this.resources = new Array();
-	this.get = function (name) {
-		for ( i = 0; i < this.resources.length; i++ )
-		{
+	get(name) {
+		for (let i = 0; i < this.resources.length; i++) {
 			var resource = this.resources[i];
```
Comment (patrikjuvonen) [inline [web]/ajax/ajax.js:145]: Prefer [for...of](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of).
```suggestion
		for (const resource of this.resources) {
```

File: [web]/elementbrowser/script.js [hunk 3]
```diff
@@ -26,10 +26,7 @@ function loadElementInfo(element)
 			var dataList = "{";
 			for ( var key in data )
 			{
-				if ( key != "toJSONString" )
-				{
-					dataList += key + " = " + data[key] + ", ";
-				}
+				dataList += key + " = " + (typeof(data[key]) === "object" ? JSON.stringify(data[key]) : data[key]) + ", ";
```
Comment (patrikjuvonen) [inline [web]/elementbrowser/script.js:29]: This for loop should be refactored to use [for...of](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of).

File: [web]/resourcemanager/manager.lua [hunk 4]
```diff
@@ -37,31 +30,3 @@ function getResourceInfo ( resource )
...
-	end
-	mta_stopResource(resource)
-end
-
-function restartResource(resource)
-	if (type(resource) == "table") then
-		resource = getResourceFromName(resource.name)
-	end
-	mta_restartResource(resource)
-end
```
Comment (patrikjuvonen) [inline [web]/resourcemanager/manager.lua:67]: By removing these functions, are we breaking backwards compatibility for the exported http functions?
https://github.com/4O4/mtasa-resources/blob/bugfix/remove-legacy-ajax-json-lib/%5Bweb%5D/resourcemanager/meta.xml#L8-L16

**Review by patrikjuvonen — CHANGES_REQUESTED** [review]
Requesting for clarification at least on the http exports part.

File: [web]/resourcemanager/manager.lua (hunk 4 above)
Comment (4O4) [reply [web]/resourcemanager/manager.lua:67]: Nope, nothing gets broken. These were just thin wrappers added to workaround the problem. Without them the exported functions are just native MTA ones (so native `startResource` and so on).

**Review by patrikjuvonen — APPROVED** [review]

**Comment by patrikjuvonen** [comment]
Thank you.

**Pull request merged**